core/src/main/java/org/jivesoftware/sparkimpl/profile/AvatarPanel.java

```diff
@@ -185,11 +191,8 @@ private void changeAvatar(final File selectedFile, final Component parent) {
             public Object construct() {
                 try {
                     ImageIcon imageOnDisk = new ImageIcon(selectedFile.getCanonicalPath());
-                    Image avatarImage = imageOnDisk.getImage();
-                    if (avatarImage.getHeight(null) > 128 || avatarImage.getWidth(null) > 128) {
-                        avatarImage = avatarImage.getScaledInstance(-1, 128, Image.SCALE_SMOOTH);
-                    }
-                    return avatarImage;
+                    setAvatar(imageOnDisk);
```

Comment:
I might simply miss something here, but why was the invocation of ``setAvatar`` moved from ``finished`` to ``construct``? Also, isn't it now colled with a larger, unscaled, image now? Was that intentional? Does that affect performance?